packages/react/src/components/FadeInOut/FadeInOut.js

```diff
@@ -0,0 +1,202 @@
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { useCallback, useEffect, useRef } from 'react';
+import { breakpoints } from '@carbon/layout';
+import PropTypes from 'prop-types';
+import settings from 'carbon-components/es/globals/js/settings';
+const { prefix } = settings;
+
+/**
+ * Amount of columns used for calculation.
+ *
+ * @private
+ */
+const _colSpan = 3;
+
+/**
+ * Utility handles fade transition for selected elements.
+ *
+ * @example
+ * import { FadeInOut } from '@carbon/ibmdotcom-react';
+ * import '@carbon/ibmdotcom-styles/scss/internal/scroll-into-view/_scroll-into-view.scss';
+ *
+ * As an example, the function can be called to target all instances of the
+ * elements in a list:
+ *
+ * const list = '.bx--content-block, .bx--content-group';
+ *
+ * For default values of 400ms and 'one and done' play:
+ * <FadeInOut selectorTargets={selectorTargets} />
+ *
+ * With 'continuous play' option:
+ * <FadeInOut selectorTargets={selectorTargets} keepAnimations={true} />
+ *
+ * For custom delay time, set within targeted class in the application's CSS code as such:
+ *
+ * .bx--content-block {
+ *   --#{$dds-prefix}--fade-in-out-delay: 250ms;
+ * }
+ *
+ */
+const FadeInOut = ({ selectorTargets, keepAnimations }) => {
+  /**
+   * The inner viewport calculation for the root margins.
+   *
+   * This calculation is done to retrieve the best fitting top and bottom
+   * margin for the fade animation to trigger/remove from elements in a
+   * user's screen.
+   *
+   * The resulting value is the optimal point where a user's attention will be
+   * grabbed by the animation without restricting their view and perception of
+   * the adopting website. The displayed elements will keep the user's attention
+   * for a longer time as they scroll down the website.
+   *
+   * @private
+   */
+  function _getViewportMargin() {
+    return (
+      '-' +
+      (
+        (document.documentElement.clientHeight * _colSpan) /
+        breakpoints.max.columns
+      ).toString() +
+      'px 0px'
+    );
+  }
+
+  /**
+   * Root margin for Inner Observer.
+   *
+   * @private
+   */
+  const _rootMargin = useRef(null);
+
+  /**
+   * Intersection Observer that watches outer viewport.
+   *
+   * @private
+   */
+  const _rootObserver = useRef(null);
+
+  /**
+   * Intersection observer that watches the inner viewport.
+   *
+   * @private
+   */
+  const _innerObserver = useRef(null);
+
+  /**
+   * Resize observer to trigger rootMargin recalculations
+   *
+   * @private
+   */
+  const _resizeObserver = useRef(null);
+
+  /**
+   * Create observers upon render and update.
+   */
+  useEffect(() => {
+    _rootObserver.current = new IntersectionObserver(handleExit);
+    _resizeObserver.current = new ResizeObserver(handleResize);
+
+    if (selectorTargets) {
+      document.querySelectorAll(selectorTargets).forEach(item => {
+        _rootObserver?.current.observe(item);
+      });
+    }
```

Comment:
`selectorTargets` search should be scoped in `<FadeInOut>`. Given React component is not automatically associated with a DOM element that's a big hard... One way to achieve this is rendering `<div>` in addition to `children`.